### PR TITLE
Incorporate use off PPP package

### LINKED ISSUE #19: Fully automated /bin scripts
As the project matures, it would be wise to fully automate the core scripts in /bin so that a user can run them without any intervention. This would ideally include pulling of the data, not just processing. This will likely require the addition of some non-R layer.

As of now, one simple issue that cannot be fully resolved is the setting of a valid working directory for the user. Attempts at using the package `here` fall short because it locks itself onto the working directory set when the library is first loaded (meaning if a user is warned they need to change their working directory and they comply, later uses of the `here()` function will not behave as they might expect, since it will still be holding on to whatever the original working directory was). Other options such as relying on .Rproj files would introduce a RStudio dependency. 

All of this can be resolved nicely by adding a simple non-R layer atop the final codebase. 

For an example of the issue: https://hrdag.org/tech-notes/harmful.html#the-working-directory-thing

### LINKED ISSUE #21: Parse dates in initial PPP read in script

### LINKED ISSUE #42: Improve NAICS matching
I am seeing evidence that some (all?) of the unmatched codes may be older, e.g., 722110: that's a valid 2007 NAICS code
it's possible different lenders use different code-years, or perhaps everyone is using old codes
It might be best to iterate through and if a match fails, go to the next newest code year list?

### PR DESCRIPTION
**Pull request by kbmorales**
Addresses the following issues:

#42 
#43 
#21 
#19 

Created an R package using code in this repo by @JohnMcCambridge and myself to download PPP data and make one clean unified PPP data set for analysis (for R users).

The package also incorporates an improved NAICS matching that iterates through subsequently older versions of NAICS index files from the US Census website.

This PR makes a script in`bin/setup.R` that will install the package and run the main assembly function in that package.

### CODE REVIEW

File: bin/setup.R [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+devtools::install_github("kbmorales/PPP")
+
+# Note: this script will download over 600MB of data into a 'data-raw'
+# folder in your home directory.
+
+# It will prompt you which version of the data you'd like to use:
+# 1 is the latest data release: 2020-08-08
+# 2 is the first data release: 2020-07-06
+
+ppp_df = PPP::ppp_assemble()
```
Comment (JohnMcCambridge) [inline bin/setup.R:20]: Got the following error (see last line). I suspect this is a relative directory issue? I just opened the .R file from the /bin directory. It created a dir in /bin (/bin/data_raw/naics) as well as in /CARES (/CARES/data-raw/All Data by State). My WD is: 
```
> getwd()
[1] "C:/Users/John/Dropbox/GitHub/DataKind-DC/CARES/bin"
```
 which got set as such because I just opened the .R file directly (and so it takes the directory of the file as WD).

As a side note, the warning "WARNING: Rtools is required to build R packages, but is not currently installed." doesn't cause any issues, I believe that is because Rtools is only needed if there is C or other compilable code, right? Might be good to add a little message reassuring people they don't need to worry?


```
> ### Assembles PPP data for further analysis
> ### kbmorales@protonmail.com
> 
> # For R users
> 
> if (!require(devtools)) install.packages('devtools')
Loading required package: devtools
Loading required package: usethis
> devtools::install_github("kbmorales/PPP")
WARNING: Rtools is required to build R packages, but is not currently installed.

Please download and install Rtools 4.0 from https://cran.r-project.org/bin/windows/Rtools/.
Skipping install of 'PPP' from a github remote, the SHA1 (01f1fc46) has not changed since last install.
  Use `force = TRUE` to force installation
> # Note: this script will download over 600MB of data into a 'data-raw'
> # folder in your home directory.
> 
> # It will prompt you which version of the data you'd like to use:
> # 1 is the latest data release: 2020-08-08
> # 2 is the first data release: 2020-07-06
> 
> ppp_df = PPP::ppp_assemble()
This function will assemble a finalized PPP dataset.
Which version of the PPP data would you like to ingest? 

1: Most recent (2020-08-08)
2: First release (2020-07-06?)

Selection: 2
This will save data to ~/data-raw/All Data by State
It will download over 100 MB of files.
Once unzipped, over 600 MB of disk space will be used.
Any existing files will be overwritten.
Proceed? 

1: Yes
2: Cancel

Selection: 1
trying URL 'https://sba.app.box.com/index.php?rm=box_download_shared_file&shared_name=tvb0v5i57oa8gc6b5dcm9cyw7y2ms6pp&file_id=f_687644633952'
Content type 'application/zip' length 105645322 bytes (100.8 MB)
downloaded 100.8 MB

Download complete.
Unzipping...
Creating path ./data-raw/
Unzip complete.
Downloading NAICS index files from US Census...
Creating path ./data-raw/naics/
Error in download.file(naics_urls[i], here::here("data-raw/naics", paste0("naics_",  : 
  cannot open destfile 'C:/Users/John/Dropbox/GitHub/DataKind-DC/CARES/data-raw/naics/naics_2017.xlsx', reason 'No such file or directory'
```

**Review by JohnMcCambridge — CHANGES_REQUESTED** [review]
see comment on setup.R

File: bin/setup.R (hunk 1 above)
Comment (kbmorales) [reply bin/setup.R:20]: Good catch. A helper function I made to create directories was the culprit--I incorporated `here::here()` into it as well to hopefully avoid this in the future. Can you reinstall the package and try again?

Re: Rtools, that will pop up the first time you use a `devtools` function, and yup you're right it just compiles R packages from source. Added a comment to `bin/setup.R` that let's people know.

**Comment by kbmorales** [comment]
Hey John just pushed changes to the `ppp` package that should resolve the wd issue. Can you test it out again and let me know if that works?

**Comment by JohnMcCambridge** [comment]
retested, seems to be working well, perhaps some edge case issues we can continue to look at, but attempts at replication have failed.
